src/main/java/io/pinecone/PineconeClientConfig.java

```diff
@@ -121,4 +137,20 @@ private String maskedApiKey() {
             return String.join("-", splits);
         }
     }
+
+    public String getUserAgent() {
+        Properties properties = new Properties();
+        String userAgentLanguage = "lang=java";
+        String version = (this.getUsageContext() != null) ?
+                userAgentLanguage + "; " + this.getUsageContext()
```

Comment:
Do we need a trailing `;` after `this.getUsageContext()` here? Is there an expected shape that needs to be adhered to when providing custom context?

Reply:
Yes, its a format we decided to follow. I'll send you the slack thread where we discussed this.